nautobot/extras/tests/test_api.py

```diff
@@ -1106,14 +1142,205 @@ def test_run_job_with_missing_data(self):
             "commit": True,
         }
 
-        url = reverse("extras-api:job-run", kwargs={"class_path": "local/api_test_job/APITestJob"})
+        url = self.get_run_url()
         response = self.client.post(url, data, format="json", **self.header)
         self.assertHttpStatus(response, status.HTTP_400_BAD_REQUEST)
         self.assertEqual(
             response.data, {"errors": {"var2": ["This field is required."], "var4": ["This field is required."]}}
         )
 
 
+class JobModelTestCase(
+    JobAPIRunTestMixin,
+    # note no CreateObjectViewTestCase
+    APIViewTestCases.GetObjectViewTestCase,
+    APIViewTestCases.ListObjectsViewTestCase,
+    APIViewTestCases.UpdateObjectViewTestCase,
+    APIViewTestCases.DeleteObjectViewTestCase,
+):
+    model = Job
+    brief_fields = ["grouping", "id", "job_class_name", "module_name", "name", "source", "url"]
+    choices_fields = None
+    update_data = {
+        # source, module_name, job_class_name, installed are NOT editable
+        "grouping_override": True,
+        "grouping": "Overridden grouping",
+        "name_override": True,
+        "name": "Overridden name",
+        "slug": "overridden-slug",
+        "description_override": True,
+        "description": "This is an overridden description.",
+        "enabled": True,
+        "approval_required_override": True,
+        "approval_required": True,
+        "commit_default_override": True,
+        "commit_default": False,
+        "hidden_override": True,
+        "hidden": True,
+        "read_only_override": True,
+        "read_only": True,
+        "soft_time_limit_override": True,
+        "soft_time_limit": 350.1,
+        "time_limit_override": True,
+        "time_limit": 650,
+    }
+    bulk_update_data = {
+        "enabled": True,
+        "approval_required_override": True,
+        "approval_required": True,
+    }
+    validation_excluded_fields = []
+
+    run_success_response_status = status.HTTP_201_CREATED
+
+    def setUp(self):
+        super().setUp()
+        job_model = Job.objects.get_for_class_path("local/api_test_job/APITestJob")
+        job_model.enabled = True
+        job_model.validated_save()
+
+    def get_run_url(self, class_path="local/api_test_job/APITestJob"):
+        job_model = Job.objects.get_for_class_path(class_path)
+        return reverse("extras-api:jobmodel-run", kwargs={"pk": job_model.pk})
+
+    def _get_detail_url(self, instance):
+        """
+        Override default implementation as we're under "jobmodel-detail" rather than "job-detail".
+        """
+        viewname = f"{self._get_view_namespace()}:jobmodel-detail"
+        return reverse(viewname, kwargs={"pk": instance.pk})
+
+    def _get_list_url(self):
+        """
+        Override default implementation as we're under "jobmodel-list" rather than "job-list".
+        """
+        viewname = f"{self._get_view_namespace()}:jobmodel-list"
+        return reverse(viewname)
```

Comment:
These rout names just gave me pause. Are these non-standard URL route names going to break some asserted patterns with helpers/utilities/templatetags/etc. where the singular is expected to match the model name?

Reply:
To be revisited as part of #1465 I think.